test/unit/tools/test_matrixtools.py

```diff
@@ -17,8 +17,8 @@ def test_is_hermitian(self):
         self.assertFalse(mt.is_hermitian(non_herm_mx))
 
     def test_is_pos_def(self):
-        pos_mx = np.array([[ 4, 0.2],
-                           [0.1, 3]], 'complex')
+        pos_mx = np.array([[ 4.0, 0.2],
+                            [0.2, 3.0]], 'complex')
```

Comment:
This matrix was not positive semidefinite, since it wasn't Hermitian.

Reply:
Makes me wonder if it was actually intended for it to be PD, not PSD in that case... Although the implementation was clearly PSD. Either way, I'm happy for the test to actually test the PSD edge case, although perhaps a version of the test with a PD matrix hitting the `attempt_cholesky` codepath could be worth adding?

Reply:
I'm not worried about testing the various PSD vs PD codepaths. I've written functions like ``is_pos_def`` a dozen times. It comes up so often with cvxpy that I actually wrote all this code:
https://github.com/cvxpy/cvxpy/blob/19bd785ae430d5e81aaf1cea3b52812edc21cc56/cvxpy/utilities/linalg.py#L50-L257.

The test matrix was neither PSD nor PD simply because being Hermitian is part of the (far and away most common) definition of PSD/PD-ness.